Dashboard/app/js/lib/components/devices/DevicesTab.jsx

```diff
@@ -0,0 +1,385 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DevicesGroupList from './DevicesGroupList';
+import DevicesList from './DevicesList';
+import RemoveDialog from './deviceTabDialog/RemoveDialog';
+import DevicesTabContext from './devices-context';
+import TABLE_NAMES from './constants';
+import AddToGroupDialog from './deviceTabDialog/AddToGroupDialog';
+import DeleteGroupDialog from './deviceTabDialog/DeleteGroupDialog';
+
+export default class DevicesTab extends React.Component {
+  state = {
+    currentTable: false,
+    devices: this.props.devices,
+    devicesGroup: this.props.devicesGroup,
+    sortAscending: false,
+    selectedColumn: null,
+    selectedDeviceIds: [],
+    selectedDevices: [],
+    selectedEditGroupId: null,
+    dialogGroupSelection: null,
+    showAddToGroupDialogBool: null,
+    showRemoveFromGroupDialogBool: false,
+    groupToEditId: null,
+    newName: null,
+  };
+
+  componentDidMount() {
+    if (this.state.devicesGroup.length === 0) {
+      setTimeout(() => {
+        this.setState({
+          devicesGroup: FLOW.deviceGroupControl
+            .get('content')
+            .getEach('_data')
+            .getEach('attributes')
+            .filter(value => Object.keys(value).length !== 0),
+        });
+      }, 500);
+    }
+  }
+
+  setCurrentTable = tableName => {
+    this.setState({ currentTable: tableName });
+    this.setState({ selectedEditGroupId: null });
+    this.setState({ newName: null });
+  };
+
+  // DEVICES LIST
+  selectDevice = (id, deviceIds) => {
+    if (deviceIds.some(deviceId => id === deviceId)) {
```

Comment:
```suggestion
    if (deviceIds.includes(id)) {
```
A small suggestion to clean this up?